packages/devextreme/js/ui/collection/ui.collection_widget.base.d.ts

```diff
@@ -21,6 +21,22 @@ import Widget, {
 
 export type ItemLike = string | CollectionWidgetItem | any;
 
+/**
+ * @docid
+ * @hidden
+ */
+export interface SelectionChangingInfo<TItem extends ItemLike = any> {
+    /** @docid */
+    readonly addedItems: Array<TItem>;
+    /** @docid */
+    readonly removedItems: Array<TItem>;
+    /**
+     * @docid
+     * @type boolean|Promise<boolean>
+     */
+    cancel?: boolean | PromiseLike<boolean>;
```

Comment:
Hm, why here do u add cancel directly to the `SelectionChangingInfo` but not in other places like [this](https://github.com/DevExpress/DevExtreme/pull/27954/files#diff-9c18fcbc8032a632fd946d886bf07de3d1d13f7b10f6f35bb5a90d1fcdeebe9cR177)?

Reply:
It's extracted into a separate interface, AsyncCancelable. We shouldn't consider it as extra data about removed and added items